src/lucky/html_builder.cr

```diff
@@ -45,8 +45,11 @@ module Lucky::HTMLBuilder
 
   macro generate_needy_initializer
     {% if !@type.abstract? %}
+      {% sorted_assigns = ASSIGNS.sort_by { |dec|
+           dec.type.resolve.nilable? || dec.value || dec.value == nil || dec.value == false ? 1 : 0
```

Comment:
I think this will work without the resolve now. That’d also be preferable because resolve can lead to weird loading issues like we used to have in models where you had to require models depending on alphabetic order

Reply:
Actually maybe not...since that would break the type union when it’s nilable with no default. So I guess this has to stay :+1: my bad